### ISSUE TITLE
ShortVariable rule doesn't work for closures directly as arguments

### ISSUE THREAD

**Issue by Orrison** [phpmd/phpmd#1239]
- PHPMD version: 2.15.0
- PHP Version: 8.4.8
- Installation type: composer
- Operating System / Distribution & Version: Ubuntu 24.04.2 LTS

### Current Behavior
The ShortVariable rule does not detect too short of variables that are direct arguments of closures passed into other functions.

### Expected Behavior
It is expected that it would detect too short of variables, even if the closure is defined directly as an argument.

### Steps To Reproduce:
This snippet:
```php
<?php

class Test
{
    public function acceptsCallback(callable $callback)
    {
        $a = 'test'; // Gets detected

        return $callback('hi');
    }

    public function calls()
    {
        $this->acceptsCallback(function ($b) { // Does NOT get detected
            return $b;
        });

        $closureDefined = function ($c) { // Gets detected
            return $c;
        };

        $this->acceptsCallback($closureDefined);
    }
}

```

Checked with:
```bash
./vendor/bin/phpmd test.php ansi naming
```

Results in:
```bash
----------------------------------------
 7  | VIOLATION | Avoid variables with short names like $a. Configured minimum length is 3.
 18 | VIOLATION | Avoid variables with short names like $c. Configured minimum length is 3.

Found 2 violations and 0 errors in 288ms
```

But it is missing detecting it in-line on line 14 as noted in the comments of the snippet.

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".
